meta-iot2000-example/recipes-support/swupdate/swupdate/swupdate_handlers.lua

```diff
@@ -115,30 +116,17 @@ function kernel_handler(image)
         return 1
     end
 
-    -- Now generate a list to update the bootloader environment
-    local tmpdirscripts = swupdate.tmpdirscripts()
-    local envfilepath = tmpdirscripts .. envfilename
-    swupdate.debug("Using " .. envfilepath .. " for setting the environment")
 
-    local envvarlist = assert(io.open(envfilepath, "w"), "Cannot create environment script")
-    envvarlist:write("kernelfile\t" .. bootlabels[rootindex] .. kernelname .. "\n")
+    -- update the bootloader
     kernelparams = kernelparams:gsub("REPLACEME", newroot)
-    envvarlist:write("kernelparams\t" .. kernelparams .. "\n")
-    envvarlist:close()
+    kernelfile = bootlabels[rootindex] .. kernelname
 
-    swupdate.debug("New environment values: ")
-    envvarlist = assert(io.open(envfilepath, "r"), "Cannot read generated environment script")
-    for value in envvarlist:lines() do
-        swupdate.debug(">>\t" .. value)
-    end
-    envvarlist:close()
+    swupdate.info(string.format("Setting bootloader environment: %s=%s", "kernelfile", kernelfile))
```

Comment:
You don't need to call `string.format()` explicitly anymore since my [commit](https://github.com/sbabic/swupdate/commit/ad25dc066880d0d79f66737685c83003597c018f) from Feb 2, 2018, as that's done under the hood. Doesn't strictly hurt though.